lib/cocoapods/generator/prepare_artifacts_script.rb

```diff
@@ -0,0 +1,246 @@
+
+      # @!group Private Helpers
+
+      # @return [String] The contents of the prepare artifacts script.
+      #
+      def script
+        script = <<-SH.strip_heredoc
+          #!/bin/sh
+          set -e
+          set -u
+          set -o pipefail
+
+          function on_error {
+            echo "$(realpath -mq "${0}"):$1: error: Unexpected failure"
+          }
+          trap 'on_error $LINENO' ERR
+
+          if [ -z ${FRAMEWORKS_FOLDER_PATH+x} ]; then
+            # If FRAMEWORKS_FOLDER_PATH is not set, then there's nowhere for us to copy
+            # frameworks to, so exit 0 (signalling the script phase was successful).
+            exit 0
+          fi
+
+          # This protects against multiple targets copying the same framework dependency at the same time. The solution
+          # was originally proposed here: https://lists.samba.org/archive/rsync/2008-February/020158.html
+          RSYNC_PROTECT_TMP_FILES=(--filter "P .*.??????")
+
+          ARTIFACT_LIST_FILE="${BUILT_PRODUCTS_DIR}/cocoapods-artifacts-${CONFIGURATION}.txt"
+          cat > $ARTIFACT_LIST_FILE
+
+          BCSYMBOLMAP_DIR="BCSymbolMaps"
+
+          record_artifact()
+          {
+            echo "$1" >> $ARTIFACT_LIST_FILE
+          }
+
+          install_artifact()
+          {
+            local source="$1"
+            local destination="$2"
+            local record=${3:-false}
+
+            # Use filter instead of exclude so missing patterns don't throw errors.
+            echo "rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter \\"- CVS/\\" --filter \\"- .svn/\\" --filter \\"- .git/\\" --filter \\"- .hg/\\" \\"${source}\\" \\"${destination}\\""
+            rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" "${source}" "${destination}"
+
+            if [[ "$record" == "true" ]]; then
+              artifact="${destination}/$(basename "$source")"
+              record_artifact "$artifact"
+            fi
+          }
+
+          # Copies a framework to derived data for use in later build phases
+          install_framework()
+          {
+            if [ -r "${BUILT_PRODUCTS_DIR}/$1" ]; then
+              local source="${BUILT_PRODUCTS_DIR}/$1"
+            elif [ -r "${BUILT_PRODUCTS_DIR}/$(basename "$1")" ]; then
+              local source="${BUILT_PRODUCTS_DIR}/$(basename "$1")"
+            elif [ -r "$1" ]; then
+              local source="$1"
+            fi
+
+            local record_artifact=${2:-true}
+            local destination="${TARGET_BUILD_DIR}"
+
+            if [ -L "${source}" ]; then
+              echo "Symlinked..."
+              source="$(readlink "${source}")"
+            fi
+
+            install_artifact "$source" "$destination" "$record_artifact"
+
+            # Locate and install any .bcsymbolmaps if present
+            find "${source}/${BCSYMBOLMAP_DIR}/" -name "*.bcsymbolmap"|while read f; do
```

Comment:
we need a `if [ -r "${source}/${BCSYMBOLMAP_DIR}" ];` here so we don't error out when there are no bcsymbolmaps